### PR TITLE
ENG-25915 Extend library to include Nicira (OvS) extensions to Openflow

### PR DESCRIPTION
**Pull request by ravilochangs**
- Add match conditions: conjunction ID, conntrack state
- Add actions: set conjunction, conntrack commit

### CODE REVIEW

File: ofp/action_nicira.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+const (
+	NxCtFlagCommit NxCtFlags = 1 << 0
+	NxCtFlagForce  NxCtFlags = 1 << 1
+)
+
+// String returns a string representation of the action type.
+func (a ActionNxType) String() string {
+	text, ok := actionNxText[a]
+	// If action is now known just say it.
```
Comment (ericatxage) [inline ofp/action_nicira.go]: should this say 'not known' ?

File: ofp/action_nicira.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+func (a *niciraHeader) ReadFrom(r io.Reader) (int64, error) {
+	return encoding.ReadFrom(r, &a.Type, &a.Len, &a.VendorID, &a.Subtype)
+}
+
+// based on struct nx_action_conjunction in ovs/lib/ofp-actions.c
+type ActionNxConjunction struct {
+	Clause     uint8
+	NumClauses uint8
+	Id         uint32
```
Comment (ericatxage) [inline ofp/action_nicira.go]: usually I name fields like this as `ID` but I think we should go with whatever convention this particular package is using.

File: ofp/action_nicira.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	}
+	err = binary.Write(buf, binary.BigEndian, a.Zone_range)
+	if err != nil {
+		return 0, err
+	}
+	err = binary.Write(buf, binary.BigEndian, a.Recirc_id)
+	if err != nil {
+		return 0, err
+	}
+	err = binary.Write(buf, binary.BigEndian, a.Padding)
```
Comment (ericatxage) [inline ofp/action_nicira.go]: I usually just add a variable like  `var padding [3]byte` right above this line  and omit it from the `struct`. Same thing can be done in `ReadFrom`

**Comment by ericatxage** [comment]
Do we need to add Xage copyright headers on these files since we authored this portin of the library?

File: ofp/action_nicira.go (hunk 3 above)
Comment (ravilochangs) [reply ofp/action_nicira.go]: Good point, changes made

File: ofputil/match_nicira.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 Xage Security, Inc. All rights reserved.
+
+package ofputil
+
+import (
+	"github.com/sensify-security/openflow/ofp"
+)
+
+// nicira creates an Openflow Nicira extensible match of the given type.
+func nicira(t ofp.NXMType, val ofp.XMValue, mask ofp.XMValue) ofp.XM {
```
Comment (ericatxage) [inline ofputil/match_nicira.go:10]: is `mask` always nil when this function is called?

**Review by ericatxage — APPROVED** [review]

**Review by isaac-xage — APPROVED** [review]

File: ofputil/match_nicira.go (hunk 4 above)
Comment (ravilochangs) [reply ofputil/match_nicira.go:10]: no, it depends on the caller, if mask is nil all bits of the value will be used for the match

Comment (ericatxage) [reply ofputil/match_nicira.go:10]: I only see 2 invocations of this function in this PR, both pass `nil` for the 3rd argument

Comment (ravilochangs) [reply ofputil/match_nicira.go:10]: matching for packet mark can take the mask value - it has not been added yet

**Pull request merged**